File: video-service-client/src/main/java/com/boclips/videos/service/client/CreateVideoRequest.java
package com.boclips.videos.service.client;

import lombok.Builder;
import lombok.Data;
import lombok.NonNull;

import java.time.LocalDate;
import java.util.List;
import java.util.Set;

@Builder
@Data
public class CreateVideoRequest {
    private final @NonNull String providerId;
    private final @NonNull String providerVideoId;
    private final @NonNull String title;
    private final @NonNull String description;
    private final @NonNull LocalDate releasedOn;
    private final String legalRestrictions;
    private final @NonNull List<String> keywords;
    private final @NonNull VideoType videoType;
    private final @NonNull String playbackId;
    private final @NonNull PlaybackProvider playbackProvider;
    private final @NonNull Set<String> subjects;
}

File: video-service/src/main/kotlin/com/boclips/videos/service/presentation/video/CreateVideoRequest.kt
package com.boclips.videos.service.presentation.video

import java.time.LocalDate

data class CreateVideoRequest(
    val provider: String? = null,
    val providerId: String? = null,
    val providerVideoId: String? = null,
    val youtubeChannelId: String? = null,
    val title: String? = null,
    val description: String? = null,
    val releasedOn: LocalDate? = null,
    val legalRestrictions: String? = null,
    val keywords: List<String>? = null,
    val videoType: String? = null,
    val playbackId: String? = null,
    val playbackProvider: String? = null,
    val searchable: Boolean? = null,
    val analyseVideo: Boolean = true,
    val ageRangeMin: Int? = null,
    val ageRangeMax: Int? = null,
    val subjects: Set<String>? = null
)
